### ISSUE TITLE
reveal.js: Explicitly setting boolean options to `false` fails

### ISSUE THREAD

**Issue by salim-b** [jgm/pandoc#7402]
## Problem

Since Pandoc 2.14.0.3 (more specifically https://github.com/jgm/pandoc/commit/14b2eb2aeb21b3dee7d07c7348ebd2c586d6a866), all [reveal.js config options](https://revealjs.com/config/) are explicitly set to their default values. 

While leaving the default of  a boolean variable untouched (1) or flipping it to `true` (2) works fine, setting it to `false` (3) fails.

### 1

```sh
echo "---
title: 'Test'
---

# My presentation

blabla
" | pandoc --to=revealjs --from=markdown --self-contained | grep mouseWheel
```

```
        mouseWheel: false,
```

### 2

```sh
echo "---
title: 'Test'
mouseWheel: true
---

# My presentation

blabla
" | pandoc --to=revealjs --from=markdown --self-contained | grep mouseWheel
```

```
        mouseWheel: true,
```

### 3

```sh
echo "---
title: 'Test'
mouseWheel: false
---

# My presentation

blabla
" | pandoc --to=revealjs --from=markdown --self-contained | grep mouseWheel
```

```
        mouseWheel: ,
```

## Workaround

Explicitly setting the boolean `false` as a string works:

```sh
echo "---
title: 'Test'
mouseWheel: 'false'
---

# My presentation

blabla
" | pandoc --to=revealjs --from=markdown --self-contained | grep mouseWheel
```

```
        mouseWheel: false,
```

## Pandoc version
2.14.0.3

**Comment by jgm** [comment]
The root problem is that the template engine (doctemplates) wasn't designed with boolean values in mind.
True gets rendered as "true", but False gets rendered as "".  Why?  Because the way conditionals work is that the condition is evaluated and tested for emptiness (empty = false, nonempty = true).

See jgm/doctemplates#16 -- ultimately we need to sort this out there.
In the mean time, it might be possible to change the writer so that it rewrites booleans as strings for these values (ugly).

**Comment by jgm** [comment]
I guess it could work to change the template so we have things like
```
mouseWheel: $if(mouseWheel)$true$else$false$endif$,
```
and similarly for other values that expect booleans.

**Comment by jgm** [comment]
One solution would be to

- change the HTML writer so that the default values are real booleans (instead of strings 'true', 'false' as now)
- change the templates like

```
mouseWheel: $if(mouseWheel)$true$else$false$endif$,
```

This is kind of ugly, but more straightforward than the present approach.

**Comment by jgm** [comment]
Of course, that approach would break with the workaround above.

**Comment by jgm** [comment]
Another possibility would be to add a "pipe" to doctemplates that converts a boolean value to the string `true` or `false`, leaves strings the same, and (does what?) with other types?
Or a more general pipe that turns a value into its canonical JSON representation?

**Comment by salim-b** [comment]
> One solution would be to
> 
>     * change the HTML writer so that the default values are real booleans (instead of strings 'true', 'false' as now)
> 
>     * change the templates like
> 
> 
> ```
> mouseWheel: $if(mouseWheel)$true$else$false$endif$,
> ```
> 
> This is kind of ugly, but more straightforward than the present approach.

I guess this makes sense until underlying issue jgm/doctemplates#16 is resolved, thus see #7403.

**Comment by jgm** [comment]
I've pushed commit 7dcf7273828a26f2f79d447ea2662ff6ef165903
which uses a tweaked version of doctemplates.  This seems to fix the problem.
(Of course, there's always the danger that it will cause other problems or break something for someone who was relying on false values rendering as empty strings...)
